bitflags.h

```diff
@@ -33,59 +35,135 @@ namespace gcpp {
 	//----------------------------------------------------------------------------
 
 	class bitflags {
+		using unit = unsigned int;
+		static_assert(std::is_unsigned<unit>::value, "unit must be an unsigned integral type.");
+
+		std::unique_ptr<unit[]> bits;
 		const int size;
-		std::vector<byte> bits;
+
+		static constexpr auto bits_per_unit = static_cast<int>(sizeof(unit) * CHAR_BIT);
+
+		//  Return a unit with all bits set if "set" is true, or all bits cleared otherwise.
+		//
+		static constexpr unit all_bits(bool set) {
+			return set ? ~unit(0) : unit(0);
+		}
+
+		//  Return a mask that will select the bit at position from its unit
+		//
+		static unit bit_mask(int at) {
+			Expects(0 <= at && "position must be non-negative");
+			return unit(1) << (at % bits_per_unit);
+		}
+
+		//  Return the number of units needed to represent a number of bits
+		//
+		static int unit_count(int bit_count) {
+			Expects(0 <= bit_count && "bit_count must be non-negative");
+			return (bit_count + bits_per_unit - 1) / bits_per_unit;
+		}
+
+		//  Get the unit that contains the bit at position
+		//
+		unit& bit_unit(int at) {
+			Expects(0 <= at && "position must be non-negative");
+			return bits[at / bits_per_unit];
+		}
+		const unit& bit_unit(int at) const {
+			Expects(0 <= at && "position must be non-negative");
+			return bits[at / bits_per_unit];
+		}
 
 	public:
-		bitflags(int bits, bool value)
-			: size{ bits }
-			, bits(1 + size / sizeof(byte), value ? byte(0xFF) : byte(0x00))
-		{ 
-			Expects(bits > 0 && "#bits must be positive");
+		bitflags(int nbits, bool value)
+			: size{ nbits }
+		{
+			Expects(nbits >= 0 && "#bits must be non-negative");
+			bits = std::make_unique<unit[]>(unit_count(nbits));
+			if (value) {
+				set_all(true);
+			}
 		}
 
 		//	Get flag value at position
 		//
 		bool get(int at) const {
 			Expects(0 <= at && at < size && "bitflags get() out of range");
-			return (bits[at / sizeof(byte)] & byte(1 << (at % sizeof(byte)))) > byte(0);
+			return (bit_unit(at) & bit_mask(at)) != unit(0);
 		}
 
 		//	Set flag value at position
 		//
 		void set(int at, bool value) {
 			Expects(0 <= at && at < size && "bitflags set() out of range");
 			if (value) {
-				bits[at / sizeof(byte)] |= byte(1 << (at % sizeof(byte)));
+				bit_unit(at) |= bit_mask(at);
 			}
```

Comment:
I like this elegance. At first I wondered why `bit_mask` did modulus instead of enforcing that `at` was less than `bits_per_unit`, but now I see you did that to support this usage. Thanks.